### PR TITLE
done with the changes

### PR DESCRIPTION
**Pull request by kuraiyamuna04**

### CODE REVIEW

File: task/.idea/.gitignore
Comment (Mohit-00-11) [inline task/.idea/.gitignore:1]: update gitignore

File: task/mediafiles/891426f9-1fac-4dfa-a241-eaceb2e287a4.jpg
Comment (Mohit-00-11) [inline task/mediafiles/891426f9-1fac-4dfa-a241-eaceb2e287a4.jpg:1]: mediafiles should be in gitignore

File: task/project/task/views.py [hunk 1]
```diff
@@ -1,32 +1,61 @@
...
             serializer.save()
             return Response(serializer.data)
         except Exception:
             return Response({"msg:You entered wrong data"}, status.HTTP_400_BAD_REQUEST)
+
+    def get(self, request):
+        user = request.user.id
+        task = TaskModel.objects.filter(assigned_by=user)
+        serializer = TaskSerializers(task, many=True)
+        return Response(serializer.data)
```
Comment (Mohit-00-11) [inline task/project/task/views.py]: we need to display task details, assignee details that is employee ( not only id), reporter(manager) details .

File: task/project/task/views.py [hunk 2]
```diff
@@ -1,32 +1,61 @@
...
+
+
+class UpdateTaskView(UpdateAPIView):
+    permission_classes = [IsAuthenticated, RequiredManager]
+
```
Comment (Mohit-00-11) [inline task/project/task/views.py]: We need patch method that will take single field and update a single field only.

File: task/project/db.sqlite3
Comment (aayush-kumar-codes) [inline task/project/db.sqlite3:1]: remove db file and from repo and ignore it as well.

File: task/mediafiles/509a3c68-0bc2-40e7-b4ac-e0543f4b2257.jpg
Comment (aayush-kumar-codes) [inline task/mediafiles/509a3c68-0bc2-40e7-b4ac-e0543f4b2257.jpg:1]: still image files are there

File: task/.idea/.gitignore
Comment (aayush-kumar-codes) [inline task/.idea/.gitignore:1]: unwanted data still here

File: task/project/app/models.py [hunk 3]
```diff
@@ -14,7 +14,8 @@
 def get_file_path(instance, filename):
     ext = filename.split('.')[-1]
     filename = "%s.%s" % (uuid.uuid4(), ext)
-    return os.path.join('', filename)
+    path = os.path.join('', filename)
+    return context['request'].build_absolute_uri(obj.profile_img)
```
Comment (aayush-kumar-codes) [inline task/project/app/models.py]: It is storing the file path with backend host, What if due to some issue we need to change the backend host ?

File: task/project/db.sqlite3
Comment (aayush-kumar-codes) [inline task/project/db.sqlite3:1]: remove this as well

File: task/project/task/serializers.py [hunk 4]
```diff
@@ -1,10 +1,34 @@
...
+class TaskDetailsSerializer(serializers.ModelSerializer):
+    assignee = serializers.SerializerMethodField()
+    reporter = serializers.SerializerMethodField()
+
+    class Meta:
+        model = TaskModel
+        fields = ("task", "assignee","reporter")
+
+    def get_assignee(self, obj):
+        user = {
```
Comment (aayush-kumar-codes) [inline task/project/task/serializers.py]: it's a kind of duplicate code as we are using the same response for the user and reporter.

Comment (aayush-kumar-codes) [reply task/project/task/serializers.py]: And each time for every single field we are fetching from 'obj.{user}.{profile}.e=xyz field.'

File: task/project/task/serializers.py [hunk 5]
```diff
@@ -1,10 +1,37 @@
...
+    assignee = serializers.SerializerMethodField()
+    reporter = serializers.SerializerMethodField()
+
+    class Meta:
+        model = TaskModel
+        fields = ("task", "assignee", "reporter")
+
+    def get_assignee(self, obj):
+        obj = obj.assigned_by
+        data = UserProfile.objects.get(user=obj)
```
Comment (aayush-kumar-codes) [inline task/project/task/serializers.py]: we have user instance already no need to make extra query here.
data = UserProfile.objects.get(user=obj)

File: task/project/task/views.py [hunk 6]
```diff
@@ -1,32 +1,83 @@
...
         except Exception:
-            return Response({"msg:You entered wrong data"}, status.HTTP_400_BAD_REQUEST)
+            return Response(no_data, status.HTTP_400_BAD_REQUEST)
+
+
+class PatchUpdateView(APIView):
+    permission_classes = [IsAuthenticated, RequiredManager | RequiredAdmin]
+
+    def patch(self, request, pk):
+        task = TaskModel.objects.get(assigned_to=pk)
```
Comment (aayush-kumar-codes) [inline task/project/task/views.py]: not using proper error handling, what if someone put wrong id ?

**Pull request merged**